### PR TITLE
[Merged by Bors] - chore: remove pointless @[reducible] attributes

### PR DESCRIPTION
**Pull request by kim-em**
This is a backport of changes made by @Ruben-VandeVelde. These attributes had no effect, and will become illegal on v4.9.0.

### CODE REVIEW

File: Mathlib/Logic/Equiv/TransferInstance.lean [hunk 1]
```diff
@@ -135,7 +135,7 @@ noncomputable instance [Small.{v} α] (R : Type*) [SMul R α] : SMul R (Shrink.{
   (equivShrink α).symm.smul R
 
 /-- Transfer `Pow` across an `Equiv` -/
-@[to_additive (attr := reducible) existing smul]
+@[reducible, to_additive existing smul]
```
Comment (semorrison) [inline Mathlib/Logic/Equiv/TransferInstance.lean:138]: This was adding the reducible attribute a second time, which will become illegal in v4.9.0 (as a sanity check).

Comment (eric-wieser) [reply Mathlib/Logic/Equiv/TransferInstance.lean:138]: I don't think either the old or the new spelling works correctly:
```
import Mathlib

@[to_additive (attr := reducible)]
def one := 1

example : one = 1 := by with_reducible rfl
example : zero = 0 := by with_reducible rfl


@[reducible, to_additive]
def one' := 1

example : one' = 1 := by with_reducible rfl
example : zero' = 0 := by with_reducible rfl
```

Comment (eric-wieser) [reply Mathlib/Logic/Equiv/TransferInstance.lean:138]: I think the correct fix is `attribute [reducible] Equiv.smul` on the line below, maybe with a TODO about `to_additive`.

File: Mathlib/Algebra/Group/Aut.lean [hunk 2]
```diff
@@ -37,8 +37,6 @@ def MulAut (M : Type*) [Mul M] :=
 #align mul_aut MulAut
 #align add_aut AddAut
 
-attribute [reducible] AddAut
```
Comment (eric-wieser) [inline Mathlib/Algebra/Group/Aut.lean:40]: I suspect this one is still needed?

File: Mathlib/Logic/Equiv/TransferInstance.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Logic/Equiv/TransferInstance.lean:138]: Note that `smul` is already defined several lines above, and has `reducible` there already.

File: Mathlib/Algebra/Group/Aut.lean (hunk 2 above)
Comment (semorrison) [reply Mathlib/Algebra/Group/Aut.lean:40]: Okay, I agree this one needs to be changed.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**